src/NodaTime.Web/Models/TimerCache.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading;
+
+namespace NodaTime.Web.Models
+{
+    public class TimerCache<T>
+    {
+        private readonly Timer timer;
+        private readonly Func<T> provider;
+        private readonly ILogger logger;
+        private readonly object padlock = new object();
+        private T value;
+        // TODO: Use Interlocked or similar to ensure proper volatility.
+        // It's possible that using volatile would be enough...
+        public T Value
+        {
+            get
+            {
+                lock (padlock)
+                {
+                    return value;
+                }
+            }
+            set
+            {
+                lock (padlock)
+                {
+                    this.value = value;
+                }
+            }
+        }
+
+        public TimerCache(IApplicationLifetime lifetime, Duration refreshPeriod, Func<T> provider, ILoggerFactory loggerFactory)
+        {
+            lifetime.ApplicationStopping.Register(() => timer?.Dispose());
+            logger = loggerFactory.CreateLogger(typeof(TimerCache<T>));
+            this.provider = provider;
+            timer = new Timer(Fetch, null, TimeSpan.Zero, refreshPeriod.ToTimeSpan());
```

Comment:
I had to check the documentation to realise that this was "now and every refreshPeriod". Worth a comment? (or perhaps naming the `dueTime` parameter, but even that isn't completely clear.)